src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,287 @@
+    }
+
+    /**
+     * Adds a task without deadline.
+     *
+     * @param s Task description.
+     * @return Associated message from Duke.
+     */
+    public String addTodo(String s) {
+        return this.tasks.addTodo(s);
+    }
+
+    /**
+     * Adds a task with deadline.
+     *
+     * @param s Task description.
+     * @param d Deadline in LocalDate format.
+     * @return Associated message from Duke.
+     */
+    public String addDeadline(String s, LocalDate d) {
+        return this.tasks.addDeadline(s, d);
+    }
+
+    /**
+     * Adds a task with deadline.
+     *
+     * @param s Task description.
+     * @param d Deadline in String format.
+     * @return Associated message from Duke.
+     */
+    public String addDeadline(String s, String d) {
+        return this.tasks.addDeadline(s, d);
+    }
+
+    /**
+     * Adds an event.
+     *
+     * @param s Event description.
+     * @param d Event time in LocalDate format.
+     * @return Associated message from Duke.
+     */
+    public String addEvent(String s, LocalDate d) {
+        return this.tasks.addEvent(s, d);
+    }
+
+    /**
+     * Adds an event.
+     *
+     * @param s Event description.
+     * @param d Event time in String format.
+     * @return Associated message from Duke.
+     */
+    public String addEvent(String s, String d) {
+        return this.tasks.addEvent(s, d);
+    }
+
+    /**
+     * Deletes the task from the list.
+     *
+     * @param i Index of the task to be deleted.
+     * @return Associated message from Duke.
+     */
+    public String deleteTask(int i) {
+        return this.tasks.deleteTask(i);
+    }
+
+    /**
+     * Parses a message written to the Duke bot.
+     *
+     * @param s Message to be parsed.
+     * @return Associated message from Duke.
+     */
+    public String parse(String s) {
+        return this.parser.parse(this, s, this.hasFinishedLoading());
+    }
+
+    /**
+     * Prints a list of tasks with the matching input string.
+     *
+     * @param s String to request for tasks with the matching string.
+     * @return List of tasks with matching string.
+     */
+    public String find(String s) {
+        return this.tasks.find(s);
+    }
+
+    //@@author chengda300
+    //Reused from https://se-education.org/guides/tutorials/javaFx.html
+    @Override
+    public void start(Stage stage) {
+        //Step 1. Setting up required components
+
+        //The container for the content of the chat to scroll.
+        scrollPane = new ScrollPane();
+        dialogContainer = new VBox();
+        scrollPane.setContent(dialogContainer);
+
+        userInput = new TextField();
+        sendButton = new Button("Send");
+
+        AnchorPane mainLayout = new AnchorPane();
+        mainLayout.getChildren().addAll(scrollPane, userInput, sendButton);
+
+        scene = new Scene(mainLayout);
+
+        stage.setScene(scene);
+        stage.show();
+
+        //Step 2. Formatting the window to look as expected
```

Comment:
I like the comments that help explain the underlying code.

Reply:
Actually that was in the GUI guide.